packages/cli/src/commands/pg/promote.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable complexity */
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import * as Heroku from '@heroku-cli/schema'
+import heredoc from 'tsheredoc'
+import {getAttachment, getRelease} from '../../lib/pg/fetcher'
+import pgHost from '../../lib/pg/host'
+import {PgStatus, PgDatabase} from '../../lib/pg/types'
+
+export default class Promote extends Command {
+  static topic = 'pg';
+  static description = 'sets DATABASE as your DATABASE_URL';
+  static flags = {
+    force: flags.boolean({char: 'f'}),
+    app: flags.app({required: true}),
+    remote: flags.remote(),
+  };
+
+  static args = {
+    database: Args.string({required: true}),
+  };
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Promote)
+    const {force, app} = flags
+    const {database} = args
+    const attachment = await getAttachment(this.heroku, app, database)
+    ux.action.start(`Ensuring an alternate alias for existing ${color.green('DATABASE_URL')}`)
+    const {body: attachments} = await this.heroku.get<Heroku.AddOnAttachment[]>(`/apps/${app}/addon-attachments`)
+    const current = attachments.find(a => a.name === 'DATABASE')
+    if (!current)
+      return
+    // eslint-disable-next-line eqeqeq
+    if (current.addon?.name === attachment.addon.name && current.namespace == attachment.namespace) {
+      if (attachment.namespace) {
+        ux.error(`${color.cyan(attachment.name)} is already promoted on ${color.app(app)}`)
+      } else {
+        ux.error(`${color.addon(attachment.addon.name)} is already promoted on ${color.app(app)}`)
+      }
+    }
+
+    const existing = attachments.filter(a => a.addon?.id === current.addon?.id && a.namespace === current.namespace)
+      .find(a => a.name !== 'DATABASE')
+    if (existing) {
+      ux.action.stop(color.green(existing.name + '_URL'))
+    } else {
+      // The current add-on occupying the DATABASE attachment has no
+      // other attachments. In order to promote this database without
+      // error, we can create a secondary attachment, just-in-time.
+      const {body: backup} = await this.heroku.post<Heroku.AddOnAttachment>('/addon-attachments', {
+        body: {
+          app: {name: app},
+          addon: {name: current.addon?.name},
+          namespace: current.namespace,
+          confirm: app,
+        },
+      })
+      ux.action.stop(color.green(backup.name + '_URL'))
+    }
+
+    if (!force) {
+      const {body: status} = await this.heroku.get<PgStatus>(`/client/v11/databases/${attachment.addon.id}/wait_status`, {
+        hostname: pgHost(),
+      })
+      if (status['waiting?']) {
+        ux.error(heredoc(`
+          Database cannot be promoted while in state: ${status.message}
+          
+          Promoting this database can lead to application errors and outage. Please run ${color.cmd('pg:wait')} to wait for database to become available.
```

Comment:
There's probably some inconsistencies around this, but when asking the user to run some other command, it's best to prepend the topic/command with `heroku`:
 
```suggestion
          Promoting this database can lead to application errors and outage. Please run ${color.cmd('heroku pg:wait')} to wait for database to become available.
```